### PR TITLE
Managing hosts adding network interfaces

### PR DESCRIPTION
**Pull request by rh-max**
#### What changes are you introducing?

Improvements to the "Adding network interfaces" section of the Managing hosts guide.

#### Why are you introducing these changes? (Explanation, links to references, issues, etc.)

https://issues.redhat.com/browse/SAT-26429

#### Checklists

* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.12/Katello 4.14 (Satellite 6.16)
* [x] Foreman 3.11/Katello 4.13 (orcharhino 6.11 on EL8 only)
* [x] Foreman 3.10/Katello 4.12
* [x] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8/6.9/6.10)
* [x] Foreman 3.8/Katello 4.10
* [x] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [x] Foreman 3.6/Katello 4.8
* [x] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* We do not accept PRs for Foreman older than 3.5.

### CODE REVIEW

File: guides/common/modules/proc_adding-a-bonded-interface-using-cli.adoc
Comment (apinnick) [inline guides/common/modules/proc_adding-a-bonded-interface-using-cli.adoc:2]:
```suggestion
= Configuring a bonded interface by using the CLI
```
"Adding" is not common in our documentation.

File: guides/common/modules/proc_adding-a-bonded-interface-using-cli.adoc
Comment (maximiliankolb) [inline guides/common/modules/proc_adding-a-bonded-interface-using-cli.adoc:1]: I would not split this off from "adding-a-bonded-interface". I see this as a feature that the procedure lists multiple ways: WebUI and CLI. Maybe we even add API or Ansible one day.

File: guides/common/modules/proc_adding-a-bmc-interface.adoc
Comment (apinnick) [inline guides/common/modules/proc_adding-a-bmc-interface.adoc:36]: AFAIK, we use numbering, not bullets, for substeps.

File: guides/common/modules/proc_adding-a-bonded-interface-using-cli.adoc
Comment (rh-max) [reply guides/common/modules/proc_adding-a-bonded-interface-using-cli.adoc:1]: So far I've only seen Foreman docs have multiple procedures in a file if they build on one another, not if they are alternatives to one another. I'll wait for a :+1: to your suggestion from another reviewer and then I'll revert this.

File: guides/common/modules/proc_adding-a-bmc-interface.adoc
Comment (rh-max) [reply guides/common/modules/proc_adding-a-bmc-interface.adoc:36]: I'll default to complying with the rest of the docs for consistency, but I'll present my logic just in case:

* The usual option results into the `a. b. c.` numbering, which is counterintuitive (looks like answer options in a test)
* It's possible to do the better `i. ii. iii.` numbering, but the markup is illogical (`.` for level one items, `...` for level two items)
* I chose bullet points, because the markup is clean enough, and the result is clear. Plus, most of the grouped steps can be done in any order, it's a form in the UI.

I'll wait for a :+1: from another reviewer to change these to `a. b. c.`.

File: guides/common/modules/proc_adding-a-bonded-interface-using-cli.adoc
Comment (maximiliankolb) [reply guides/common/modules/proc_adding-a-bonded-interface-using-cli.adoc:1]: Example: https://docs.theforeman.org/nightly/Managing_Content/index-katello.html#Creating_a_Custom_Product_content-management

File: guides/common/modules/proc_adding-a-bmc-interface.adoc
Comment (asteflova) [reply guides/common/modules/proc_adding-a-bmc-interface.adoc:36]: My point of view is based on "separating content from presentation": you want to add substeps, so let's use the markup for substeps. The way that markup is rendered is a separate issue.

File: guides/common/modules/con_network-interface-configuration-options.adoc
Comment (Lennonka) [inline guides/common/modules/con_network-interface-configuration-options.adoc:7]:
```suggestion
.Physical interface settings
```
I'd prefer informal headings to separate the options.

File: guides/common/modules/con_network-interface-configuration-options.adoc
Comment (Lennonka) [inline guides/common/modules/con_network-interface-configuration-options.adoc:29]:
```suggestion
*Tag*:: You can set a VLAN tag to trunk a network segment from the physical network through to the virtual interface.
```
Is there a reason for a different markup than the options above?

File: guides/common/modules/con_network-interface-configuration-options.adoc
Comment (Lennonka) [inline guides/common/modules/con_network-interface-configuration-options.adoc:36]:
```suggestion
*Mode*:: The bonding mode defines a policy for fault tolerance and load balancing.
```
Same here.

**Review by Lennonka — CHANGES_REQUESTED** [review]
A couple of suggestions.

**Comment by swadeley** [comment]
Hi, lgtm

**Comment by rh-max** [comment]
@maximiliankolb If you have no further feedback, could you please merge & cherrypick this? Stephen provided the technical review.

File: guides/common/modules/ref_network-interface-configuration-options.adoc
Comment (Lennonka) [inline guides/common/modules/ref_network-interface-configuration-options.adoc:18]:
```suggestion
If using installer-based provisioning, a configuration file is automatically created for the interface.
```
Kickstart is installer configuration specific to EL, but I think it applies to any OS installer. I'd go with a more generic term to make it more upstream friendly.

File: guides/common/modules/ref_network-interface-configuration-options.adoc
Comment (Lennonka) [inline guides/common/modules/ref_network-interface-configuration-options.adoc:22]:
```suggestion
*Provision*:: Network boot uses the provisioning interface.
```

**Review by Lennonka — COMMENTED** [review]
Overally looks good. I suggest two minor updates in terminology, but they should be confirmed by SMEs.

File: guides/common/modules/ref_network-interface-configuration-options.adoc
Comment (Lennonka) [reply guides/common/modules/ref_network-interface-configuration-options.adoc:18]: @maximiliankolb Can you or someone from your team confirm it applies to other OSs as well?

File: guides/common/modules/ref_network-interface-configuration-options.adoc
Comment (Lennonka) [reply guides/common/modules/ref_network-interface-configuration-options.adoc:22]: @stejskalleos Is is this change acceptable?

Comment (stejskalleos) [reply guides/common/modules/ref_network-interface-configuration-options.adoc:22]: LGTM

File: guides/common/modules/ref_network-interface-configuration-options.adoc
Comment (Lennonka) [reply guides/common/modules/ref_network-interface-configuration-options.adoc:18]: In the context of Foreman, installer-based provisioning should be considered automated. Anaconda+Kickstart is for EL, but Debian/Ubuntu uses a different installer+config. But Foreman generates configuration for Debian/Ubuntu as well, it just isn't called Kickstart.
Then perhaps a better term would be:
```suggestion
If you use provisioning with installer automation, a configuration file is automatically created for the interface.
```

Comment (Lennonka) [reply guides/common/modules/ref_network-interface-configuration-options.adoc:18]: @stejskalleos Would that ^^ work?

Comment (stejskalleos) [reply guides/common/modules/ref_network-interface-configuration-options.adoc:18]: If, in this context, we expect the installer workflow always to be automated, then LGTM.

Comment (maximiliankolb) [reply guides/common/modules/ref_network-interface-configuration-options.adoc:18]: Using a OS-neutral term would be appreciated, yes.

File: guides/common/modules/con_configuring-network-interfaces.adoc
Comment (maximiliankolb) [inline guides/common/modules/con_configuring-network-interfaces.adoc:8]:
```suggestion
When configuring an interface, select one of:
```

File: guides/common/modules/proc_configuring-a-bmc-interface.adoc
Comment (maximiliankolb) [inline guides/common/modules/proc_configuring-a-bmc-interface.adoc:1]:
```suggestion
[id="configuring-a-baseboard-management-controller-interface"]
```

If we change it, then let's go with dashes.

File: guides/common/modules/proc_configuring-a-bmc-interface.adoc
Comment (maximiliankolb) [inline guides/common/modules/proc_configuring-a-bmc-interface.adoc:32]:
```suggestion
```

File: guides/common/modules/proc_configuring-a-bmc-interface.adoc
Comment (maximiliankolb) [inline guides/common/modules/proc_configuring-a-bmc-interface.adoc:36]:
```suggestion
. In the {ProjectWebUI}, navigate to *Hosts* > *All Hosts*.
. Click *Edit* next to the host you want to edit.
. On the *Interfaces* tab, click *Add Interface*.
```

We generally try to avoid nested lists. Is that something you want to introduce on purpose? I am not against it in generally, but I like sticking to one style/to conventions. Maybe one of your colleagues can weigh in. I am OK either way and appreciate the effort.

File: guides/common/modules/con_configuring-network-interfaces.adoc
Comment (maximiliankolb) [inline guides/common/modules/con_configuring-network-interfaces.adoc:1]:
```suggestion
[id="configuring-network-interfaces"]
```

This makes `grep` even more fun because you'll find the file, anchor, and all links if you drop the "con_" prefix.

**Review by maximiliankolb — CHANGES_REQUESTED** [review]

File: guides/common/modules/proc_configuring-a-physical-interface.adoc
Comment (apinnick) [inline guides/common/modules/proc_configuring-a-physical-interface.adoc:6]: I am concerned about the number of subprocedures you are creating. The general interface, ntwork, and DNS settings could be converted to bullet lists.

I'm not sure navigating to "Add Interface" should be a subprocedure. I would make them part of the procedure.

File: guides/common/modules/proc_configuring-a-physical-interface.adoc
Comment (apinnick) [inline guides/common/modules/proc_configuring-a-physical-interface.adoc:30]: This kind of subprocedure, where each step starts with "Select whether ..." is not easy for a user to scan.

**Comment by asteflova** [comment]
@rh-max seems to have implemented feedback from two maintainers and from the Jira ticket. We can always improve things further if needed but for now, let's go ahead with the merge.

**Pull request merged**
